sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/identity/impl/utils/ClassUtil.java

```diff
@@ -0,0 +1,50 @@
+package com.azure.spring.cloud.service.implementation.identity.impl.utils;
+
+import com.azure.core.exception.AzureException;
+import com.azure.spring.cloud.service.implementation.identity.api.credential.TokenCredentialProvider;
+import com.azure.spring.cloud.service.implementation.identity.api.credential.TokenCredentialProviderOptions;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+public class ClassUtil {
+
+    private ClassUtil() {
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> T instantiateClass(Class<T> clazz, Object... args) {
+        try {
+            // TODO not use this paramater class
+            Constructor<?> ctor = clazz.getDeclaredConstructor(TokenCredentialProviderOptions.class);
+            ctor.setAccessible(true);
+            // TODO pass the args or call default ctor
+            return (T) ctor.newInstance(args);
+        } catch (InstantiationException e) {
+            // TODO
+            throw new AzureException(e);
```

Comment:
Either we resolve these todos, or we can use BeanUtils directly in this PR.

Reply:
But I'd prefer to improve this class.

Reply:
It's also my preference, will improve the class.